wee_slack.py

```diff
@@ -3659,6 +3659,19 @@ def command_openweb(data, current_buffer, args):
         url = "https://{}/archives/{}/p{}000000".format(channel.team.domain, channel.slack_name, now.majorstr())
     w.prnt_date_tags(channel.team.channel_buffer, SlackTS().major, "openweb,logger_backlog_end,notify_none", url)
 
+@slack_buffer_required
+def command_linkarchive(data, current_buffer, args):
+    channel = EVENTROUTER.weechat_controller.buffers[current_buffer]
+    message_id = args.split(' ', 1)[1]
+    if message_id[0] == "$":
+        message_id = message_id[1:]
+    if isinstance(channel, SlackChannelCommon) and message_id in channel.hashed_messages:
+        message = channel.hashed_messages[message_id]
+        ts = message.ts
+        url = "https://{}/archives/{}/p{}{:0>6}".format(channel.team.domain, channel.identifier, ts.majorstr(), ts.minorstr())
+        if isinstance(channel, SlackThreadChannel):
```

Comment:
You should rather check if the message is a thread message here, otherwise links for thread messages printed in the channel (with the thread_messages_in_channel option) won't get these parameters.

Reply:
Fixed too. I considered using the ThreadMessage and didn’t think about this possible case so I kept the ThreadChannel initially.
I’m impressed by all the subtelties you know about slack!

Reply:
Hehe, thanks :)